java/ql/src/semmle/code/xml/MavenPom.qll

```diff
@@ -485,3 +485,85 @@ class MavenRepoJar extends File {
     else getVersion().matches(pom.getVersionString() + "%")
   }
 }
+
+/**
+ * A `<plugin>` element in a Maven POM file.
+ */
+class MavenPlugin extends ProtoPom {
+  MavenPlugin() { this.hasName("plugin") }
+}
+
+/**
+ * A `maven-shade-plugin` plugin element in a Maven POM file.
+ */
+class MavenShadePlugin extends MavenPlugin {
+  MavenShadePlugin() {
+    (
+      this.getGroup().getValue() = "org.apache.maven.plugins" or
+      not exists(this.getGroup()) // org.apache.maven.plugins is the default
+    ) and
+    this.getArtifact().getValue() = "maven-shade-plugin"
+  }
+
+  MavenShadeRelocation getARelocation() { result.getPlugin() = this }
+}
+
+/**
+ * A `<relocation>` specification within a `maven-shade-plugin` configuration.
+ */
+class MavenShadeRelocation extends XMLElement {
+  MavenShadePlugin plugin;
+
+  MavenShadeRelocation() {
+    this.getParent().(XMLElement).getParent().(XMLElement).getParent() = plugin and
```

Comment:
Might be good to add a comment explaining the hierarchy here, i.e. `plugin -> configuration -> relocations -> relocation`

It is also possible to specify the configuration per execution, as it was done in the [example of the Maven Shade Plugin](https://maven.apache.org/plugins/maven-shade-plugin/examples/class-relocation.html), see also the [POM Reference](https://maven.apache.org/pom.html#Plugins).